protocols/stream/src/behaviour.rs

```diff
@@ -134,7 +171,10 @@ impl NetworkBehaviour for Behaviour {
         if let Poll::Ready(Some(peer)) = self.dial_receiver.poll_next_unpin(cx) {
             return Poll::Ready(ToSwarm::Dial {
                 opts: DialOpts::peer_id(peer)
-                    .condition(PeerCondition::DisconnectedAndNotDialing)
+                    .condition(match self.connection_policy {
+                        ConnectionPolicy::AcceptAll => PeerCondition::DisconnectedAndNotDialing,
+                        ConnectionPolicy::IgnoreRelayed => PeerCondition::Always,
+                    })
```

Comment:
Let's add a comment on why this is changed for `ConnectionPolicy::IgnoreRelayed`.